### ISSUE TITLE
Non Linearity in output layer stops learning

### ISSUE THREAD

**Issue by hhsecond** [joelgrus/fizz-buzz-tensorflow#3]
Hi Joelgrus, 
First of all, thanks a ton for your astonishing work. I was playing around with the network and found that the non-linearity in the output layer (sigmod or softmax) would stop the learning process. How much ever I train the network, its not learning anything. Cost is not going down and accuracy is not going up. Any thoughts?
